relay-server/src/utils/rate_limits.rs

```diff
@@ -378,18 +387,59 @@ where
         let mut enforcement = Enforcement::default();
 
         if let Some(category) = summary.event_category {
-            let event_limits = (self.check)(scoping.item(category), 1)?;
-            let longest = event_limits.longest();
-            enforcement.event = CategoryLimit::new(category, 1, longest);
-
-            // Record the same reason for attachments, if there are any.
-            enforcement.attachments = CategoryLimit::new(
-                DataCategory::Attachment,
-                summary.attachment_quantity,
-                longest,
-            );
-
-            rate_limits.merge(event_limits);
+            if category == DataCategory::Transaction {
+                if summary.transaction_metrics_extracted {
```

Comment:
I do not follow the reasoning behind this branch here, could you elaborate? Even if there is a transaction item without metrics, then we still need to invoke the rate limiter for the (indexed) transaction category.

Reply:
If transactions are extracted then we're at the `enforce_quotas()` part of the flow and we need to execute that logic.  If not we're at the `check_envelope` part.  There is the case where an earlier relay has already extracted the metrics and we go straight to the enforce_quotas behaviour even in check_envelope.